### ISSUE TITLE
poor val_acc

### ISSUE THREAD

**Issue by HIN0209** [junkwhinger/grad_cam#1]
Hello, this is a great project. Although both ipynb files worked, I have an issue on discrepancy of val_acc from yours.

Setting: Ubuntu16.04, anaconda3, python3.5, Keras2.1.5, Tensorflow-gpu1.6.0

I downloaded the suggested dataset and run resnet_cam.ipynb. It was the same as shown in github until finding images. Then, when training 50 epochs, below is the output:

Epoch 49/50
45/46 [============================>.]45/46 [============================>.] - ETA: 0s - loss: 0.2111 - acc: 0.9285
Epoch 00049: val_acc did not improve
46/46 [==============================]46/46 [==============================] - 12s 272ms/step - loss: 0.2134 - acc: 0.9280 - val_loss: 4.4084 - val_acc: 0.1812

Epoch 50/50
45/46 [============================>.]45/46 [============================>.] - ETA: 0s - loss: 0.2281 - acc: 0.9250
Epoch 00050: val_acc did not improve
46/46 [==============================]46/46 [==============================] - 12s 269ms/step - loss: 0.2299 - acc: 0.9246 - val_loss: 4.0736 - val_acc: 0.1974
-----------------------------------------------------
So, it was obviously over-fitting and the val_acc was around 0.2. How did you achieve the 0.87 val_acc as provided?

**Comment by junkwhinger** [comment]
Hi, first of all, thank you for the comment.

I've just run my code on my local machine with 1060ti GPU.
Setting: Windows10, anaconda3, python3.6, TensorFlow-GPU 1.4.0 (don't need Keras here)
and the following is my intermediate result.

Epoch 24/50
45/46 [============================>.] - ETA: 0s - loss: 0.3134 - acc: 0.9077Epoch 00023: val_acc improved from 0.70625 to 0.76875, saving model to weights-improvement-23-0.77.hdf5
46/46 [==============================] - 15s - loss: 0.3124 - acc: 0.9083 - val_loss: 0.6401 - val_acc: 0.7688

I changed the input data path, and the rest stayed the same.
It was very peculiar to me that the validation loss soared for the first 10 ~ 12 epochs when the training loss decreased rapidly. Then around 12 to 14 epochs as the validation loss reduced the accuracy started improving.

I have been playing with PyTorch with the same dataset and the same architecture, and didn't run into this strange learning pattern. It might have to do with the constant learning rate, tho. Hope this answer helps.

**Comment by HIN0209** [comment]
@junkwhinger 
Thank you for your update. Yes, the learning rate should be the first to suspect and I will try fine-tuning.
